auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -30,25 +30,35 @@
     @Nullable private final String mEmail;
     private final String mToken;
     private final String mSecret;
+    private final String mPrevUid;
 
     public IdpResponse(String providerId, @Nullable String email) {
-        this(providerId, email, null, null);
+        this(providerId, email, null, null, null);
     }
 
-    public IdpResponse(
-            String providerId, @Nullable String email, @Nullable String token) {
-        this(providerId, email, token, null);
+    public IdpResponse(String providerId, @Nullable String email, @Nullable String token) {
+        this(providerId, email, token, null, null);
     }
 
     public IdpResponse(
             String providerId,
             @Nullable String email,
             @Nullable String token,
-            @Nullable String secret) {
+            @Nullable String secret,
+            @Nullable String prevUid) {
         mProviderId = providerId;
         mEmail = email;
         mToken = token;
         mSecret = secret;
+        mPrevUid = prevUid;
+    }
+
+    public IdpResponse(IdpResponse response, String prevUid) {
+        this(response.getProviderType(),
+             response.getEmail(),
+             response.getIdpToken(),
+             response.getIdpSecret(),
+             prevUid);
```

Comment:
This is a mess, but I don't know how to get around it. I need to add the previous user id to an existing `IdpResponse` and I choose not to use a setter since this is public api. Any thoughts on a better way to do this? Should I be using a setter? Also, if we stick with this, what is the convention for this kind of constructor? (`response.getProviderType()` vs `response.mProviderId`)